pandas/core/groupby/generic.py

```diff
@@ -234,9 +234,16 @@ def aggregate(
         relabeling = func is None
         columns = None
         no_arg_message = "Must provide 'func' or named aggregation **kwargs."
+        tuple_given_message = "func is expected but recieved {} in **kwargs."
         if relabeling:
             columns = list(kwargs)
```

Comment:
Don't need this anymore.

Reply:
Yes, Agreed. `tuple_given_message` is same as `no_arg_message`. How about `no_arg_message = "Must provide 'func' or named aggregation **kwargs but recieved {}."`?

Reply:
This comment was specifically about the `columns = list(kwargs)`, since you're building up `func` differently. But my suggestion may have been incorrect. `columns` might be used later on.

Reply:
Yes, it is used.
```
if not columns:
    raise TypeError(no_args_message)
```